pallets/xyk/src/lib.rs

```diff
@@ -1940,11 +1987,14 @@ impl<T: Config> XykFunctionsTrait<T::AccountId> for Pallet<T> {
 			second_asset_amount.into(),
 			ExistenceRequirement::KeepAlive,
 		)?;
-		Pallet::<T>::set_liquidity_minting_checkpoint(
-			sender.clone(),
-			liquidity_asset_id,
-			liquidity_assets_minted,
-		)?;
+		// Liquidity minting functions not triggered on not promoted pool
+		if <T as Config>::PoolPromoteApi::get_pool_rewards(liquidity_asset_id).is_some() {
```

Comment:
looks like we never read value returned by get_pool_rewards, the only thing we are interested is if it is some or none. Then api should be modified to only expose that boolean information

Reply:
we use the value at xyk 687:
let available_rewards_for_pool: U256 = U256::from(
			<T as Config>::PoolPromoteApi::get_pool_rewards(liquidity_asset_id)
				.ok_or_else(|| DispatchError::from(Error::<T>::NotAPromotedPool))?,
		);